### ISSUE TITLE
Bug: [no-deprecated] Exported classes implementing a deprecated interface are not reported

### ISSUE THREAD

**Issue by wgoehrig** [typescript-eslint/typescript-eslint#10253]
### Before You File a Bug Report Please Confirm You Have Done The Following...

- [X] I have tried restarting my IDE and the issue persists.
- [X] I have updated to the latest version of the packages.
- [X] I have [searched for related issues](https://github.com/typescript-eslint/typescript-eslint/issues?q=is%3Aissue+label%3A%22package%3A+eslint-plugin%22) and found none that matched my issue.
- [X] I have [read the FAQ](https://typescript-eslint.io/linting/troubleshooting) and my problem is not listed.

### Playground Link

https://typescript-eslint.io/play/#ts=5.6.2&fileType=.tsx&code=PQKhAIAEBMFMAcBOsDGBDALra4TAFACWAdlogGZoqzgAKsG4A3gL774oA2aAzj%2BAGFM4QgFt4nWKNil%2B9Rq3ywAHvAD2iRl178AImoDmI8ZOmy6DZmyA&eslintrc=N4KABGBEBOCuA2BTAzpAXGUEKQAIBcBPABxQGNoBLY-AWhXkoDt8B6Jge1oBNFjpEZAIb5E3dFETRoHaJHBgAviEVA&tsconfig=N4KABGBEDGD2C2AHAlgGwKYCcDyiAuysAdgM6QBcYoEEkJemy0eAcgK6qoDCAFutAGsylBm3TgwAXxCSgA&tokens=false

### Repro Code

```TypeScript
/** @deprecated */
interface Pet {}

class Cat implements Pet {}
export class Dog implements Pet {}
```

### ESLint Config

```javascript
module.exports = {
  parser: "@typescript-eslint/parser",
  rules: {
    "@typescript-eslint/no-deprecated": "error",
  },
};
```

### tsconfig

```jsonc
{
  "compilerOptions": {
    "strictNullChecks": true
  }
}
```

### Expected Result

I expected that line 5 should report the error "`Pet` is deprecated.", just as it does (correctly) for line 4.  I expect use of a deprecated interface to be flagged regardless of whether the class implementing it is exported or not.

### Actual Result

No error is reported for line 5 unless I remove the the `export` keyword.

**Comment by kirkwaiblinger** [comment]
Thanks for the nice and concise repro! 👌
